### PR TITLE
panic handler: don't panic if DEBUG_WRITER is not set

### PR DESCRIPTION
**Pull request by lschuermann**
### Pull Request Overview

During development of a new board one might panic prior to a debug
writer component being registered. Especially if no serial console is
working yet, seeing the blinking LEDs in the recognizable pattern is a
good first indicator.

If no debug writer component is set, the panic handler will itself
panic, resulting in a panic loop. The LED pattern will not be
displayed.

This checks whether the DEBUG_WRITER global variable is set and prints
a message that it should be set with `set_debug_writer_wrapper`
instead.

### Testing Strategy

This pull request was tested by running on an in-development board that does not call to `set_debug_writer_wrapper`.

### TODO or Help Wanted
N/A

### Documentation Updated

- N/A Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [ ] Ran `make prepush`.

### CODE REVIEW

**Review by bradjc — DISMISSED** [review]

**Review by hudson-ayers — APPROVED** [review]

File: kernel/src/debug.rs [hunk 1]
```diff
@@ -343,10 +343,12 @@ pub struct DebugWriter {
 /// needed so the debug!() macros have a reference to the object to use.
 static mut DEBUG_WRITER: Option<&'static mut DebugWriterWrapper> = None;
 
+pub unsafe fn try_get_debug_writer() -> Option<&'static mut DebugWriterWrapper> {
```
Comment (ppannuto) [inline kernel/src/debug.rs:346]: I don't think this should be a `pub` function. I see that this mirrors the non-try version, but that really should not be `pub` either.. 

The underlying `DEBUG_WRITER` is pretty unsafe (and not a `pub` object). Constraining its access to this file seems valuable.

--

Could you remove `pub` here and while we're at it from the non-try variant?

**Review by ppannuto — CHANGES_REQUESTED** [review]

File: kernel/src/debug.rs (hunk 1 above)
Comment (lschuermann) [reply kernel/src/debug.rs:346]: Can do :+1:.

Comment (lschuermann) [reply kernel/src/debug.rs]: Thanks for the hint. I've pushed the changes and added the following to the commit message:

_Due to the static mutable debug writer being inherently unsafe, this also reduces its visibility._

Feel free to resolve this if you are comfortable with the result

**Review by hudson-ayers — APPROVED** [review]

**Review by alistair23 — APPROVED** [review]

**Review by ppannuto — APPROVED** [review]

**Pull request merged**